globals.h

```diff
@@ -14,38 +15,37 @@ constexpr auto kDefaultChainLength        = 8;
 constexpr auto kDefualtRows               = 32;
 constexpr auto kDefaultColumns            = 64;
 constexpr auto kDefaultGPIOSlowdown       = 5;
-constexpr auto kDefaultRefreshRate        = 60;
+constexpr auto kDefaultRefreshRate        = 100;
 constexpr auto kDefaultDisableBusyWaiting = true;
 
 constexpr auto kIncomingSocketPort        = 49152;
 constexpr auto kMaxBuffers                = 500;
 
-#define NUM_LEDS (Rows * Columns * ChainLength)
-
 // Helpers for extracting values from memory in a system-independent way
 
 constexpr uint64_t ULONGFromMemory(const uint8_t * payloadData)
 {
-    return  (uint64_t)payloadData[7] << 56  |
-            (uint64_t)payloadData[6] << 48  |
-            (uint64_t)payloadData[5] << 40  |
-            (uint64_t)payloadData[4] << 32  |
-            (uint64_t)payloadData[3] << 24  |
-            (uint64_t)payloadData[2] << 16  |
-            (uint64_t)payloadData[1] << 8   |
-            (uint64_t)payloadData[0];
+    const auto result = *(uint64_t *)payloadData;
+    if constexpr (std::endian::native == std::endian::little) 
+        return result;                      // Already little-endian, no swap needed
+    else 
+        return __builtin_bswap64(result);   // Swap for big-endian systems
```

Comment:
If you can use C++23, there's `std::byteswap` as a platform-independent alternative.